flink-ml-iteration/src/main/java/org/apache/flink/iteration/datacache/nonkeyed/Segment.java

```diff
@@ -18,61 +18,37 @@
 
 package org.apache.flink.iteration.datacache.nonkeyed;
 
-import org.apache.flink.core.fs.Path;
+import org.apache.flink.annotation.Internal;
 
-import java.io.Serializable;
-import java.util.Objects;
+/** A segment contains the information about a cache unit. */
+@Internal
+class Segment {
 
-/** A segment represents a single file for the cache. */
-public class Segment implements Serializable {
+    private FileSegment fileSegment;
 
-    private final Path path;
+    private MemorySegment memorySegment;
```

Comment:
Instead of creating a new class `MemorySegment` whose name collide with a class in `org.apache.flink.core.memory.MemorySegment`, would it be simpler to just put `List<org.apache.flink.core.memory.MemorySegment>` inside `Segment`?

The `count` in `MemorySegment` and `FileSegment` can also be moved to `Segment` if we need it.